apps/worker/src/app/workflow/services/metric-queue.service.ts

```diff
@@ -0,0 +1,125 @@
+const nr = require('newrelic');
+import { WorkerOptions } from 'bullmq';
+import { Inject, Injectable, Logger } from '@nestjs/common';
+import { QueueService } from '@novu/application-generic';
+import { min, max, sum, mean } from 'simple-statistics';
+
+interface IMetric {
+  count: number;
+  total: number;
+  min: number;
+  max: number;
+  sumOfSquares: number;
+}
+
+function sumOfSquares(values: number[]): number {
+  const meanVal = mean(values);
+
+  return sum(values.map((value) => (value - meanVal) ** 2));
+}
+
+@Injectable()
+export class MetricQueueService extends QueueService<Record<string, never>> {
```

Comment:
I'm not sure regarding this extension here from QueueService, because the QueueService previously served the job processing queue, and I'm afraid extending it like this can create a lot of problems. @LetItRock could you help with reviewing this one?

Reply:
Thank you for the explanation @LetItRock this makes sense now, appreciate it!